File: cpp/questions/bestClosingTime.cpp
class Solution {
public:
    int bestClosingTime(std::string customers) {
        std::vector<std::pair<int, int>> nm;
        int n = customers.size();
        int yc = 0;
        int nc = 0;
        int minc = 0;
        int ans = -1;
        for (int p=0; p<n; p++) {
            if (customers[p] == 'N') {
                nc++;
            } else {
                yc--;
            }
            if (minc > nc + yc) {
                minc = nc + yc;
                ans = p;
            }
        }
        return ans==-1 ? 0 : ans+1;;
    }
};

File: python/questions/bestClosingTime.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        n = len(customers)
        ans = -1
        minc = 0
        yc = 0
        nc = 0
        for p in range(n):
            if customers[p] == 'N':
                nc += 1
            else:
                yc -= 1
            if minc > nc+yc:
                minc = nc+yc
                ans = p
        if ans == -1:
            return 0
        else:
            return ans + 1
